### PR TITLE
BTAT-5783 Added mandation status predicate

### PR DESCRIPTION
**Pull request by TomDudley92**

### CODE REVIEW

File: it/pages/SubmitFormPageSpec.scala [hunk 1]
```diff
@@ -38,11 +38,15 @@ class SubmitFormPageSpec extends BaseISpec {
...
 
           AuthStub.stubResponse(OK, mtdVatAuthResponse)
-          VatSubscriptionStub.stubResponse(OK, vatSubscriptionSuccessJson)
+          VatSubscriptionStub.stubResponse("customer-details", OK, customerInformationSuccessJson)
+          VatSubscriptionStub.stubResponse("mandation-status", OK, mandationStatusSuccessJson)
           VatObligationsStub.stubResponse(OK, vatObligationsSuccessJson)
 
           val response: WSResponse = request
 
+          println()
```
Comment (miranda-hawkes) [inline it/pages/SubmitFormPageSpec.scala]: Please remove

File: test/connectors/VatSubscriptionConnectorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        val testUrl = TestVatSubscriptionConnector.vatSubscriptionUrl(vrn, "endpoint")
+        testUrl shouldBe s"${mockAppConfig.baseUrl("vat-subscription")}/vat-subscription/999999999/endpoint"
+      }
+    }
+
+    "calling 'getCustomerDetails'" when {
+
+      def result: Future[HttpGetResult[CustomerDetails]] = TestVatSubscriptionConnector.getCustomerDetails(vrn)
+
+      "called for a Right with CustomerDetails" should {
```
Comment (miranda-hawkes) [inline test/connectors/VatSubscriptionConnectorSpec.scala]: This wording is confusing as is the one on line 61 - could you reword it to be like the example on line 74 please?

File: test/mocks/MockHttp.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.mockito.MockitoSugar
+import play.api.libs.json.Writes
+import uk.gov.hmrc.http.HttpReads
+import uk.gov.hmrc.play.test.UnitSpec
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.Future
+
+trait MockHttp extends UnitSpec with MockitoSugar with BeforeAndAfterEach {
```
Comment (miranda-hawkes) [inline test/mocks/MockHttp.scala]: If we're already using scalamock for the controllers and services... could we be consistent and use it everywhere instead of mockito here please?

File: app/controllers/predicates/MandationStatusPredicate.scala [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  override def refine[A](request: User[A]): Future[Either[Result, User[A]]] = {
+
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+    implicit val req: User[A] = request
+
+
+    mandationStatusService.getMandationStatus(req.vrn) map {
+      case Right(MandationStatus(`nonMTDfB`)) => Right(request)
+      case Right(unsupportedMandationStatus) =>
+        Logger.debug(s"[MandationStatusPredicate][refine] - Incorrect mandation status returned. Status returned was: $unsupportedMandationStatus")
```
[the remark this answers is not in the document]
Comment (TomDudley92) [reply app/controllers/predicates/MandationStatusPredicate.scala]: So would you rather this to be a warn as well rather than debug?

[the remark this answers is not in the document]
Comment (miranda-hawkes) [reply app/controllers/predicates/MandationStatusPredicate.scala]: Its not something we'd care to see in production logs so debug is fine (could you remove the WARN one please)
